airflow/www/views.py

```diff
@@ -1531,22 +1539,13 @@ def graph(self, session=None):
                 }
             })
 
-        def get_upstream(task):
-            for t in task.upstream_list:
-                edge = {
-                    'u': t.task_id,
-                    'v': task.task_id,
-                }
-                if edge not in edges:
-                    edges.append(edge)
-                    get_upstream(t)
-
-        for t in dag.roots:
-            get_upstream(t)
-
-        dt_nr_dr_data = get_date_time_num_runs_dag_runs_form_data(request, session, dag)
-        dt_nr_dr_data['arrange'] = arrange
-        dttm = dt_nr_dr_data['dttm']
+        for edge in session.query(models.DagEdge)\
```

Comment:
Just a style question, but WDYT about expressing this as a comprehension like `task_instances_dict` and `ops` below for consistency?